ceph/ceph_admin/helper.py

```diff
@@ -545,20 +545,23 @@ def generate_rgw_spec(self, spec):
         if node_names:
             spec["placement"]["hosts"] = self.get_hostnames(node_names)
 
-        if spec["spec"].get("rgw_frontend_ssl_certificate") == "create-cert":
-            subject = {
-                "common_name": spec["placement"]["hosts"][0],
-                "ip_address": self.cluster.get_node_by_hostname(
-                    spec["placement"]["hosts"][0]
-                ).ip_address,
-            }
-            key, cert, ca = generate_self_signed_certificate(subject=subject)
-            pem = key + cert + ca
-            cert_value = "|\n" + pem
-            spec["spec"]["rgw_frontend_ssl_certificate"] = "\n    ".join(
-                cert_value.split("\n")
-            )
-            LOG.debug(pem)
+        if spec.get("spec", False):
```

Comment:
@anrao19 
whats the problem with the existing code here
could you add it in the description as well

Reply:
@hmaheswa , if we use "command: apply_spec" without key "spec" it was failling with key error spec